api/datadoghq/v2alpha1/datadogpodautoscaler_types.go

```diff
@@ -0,0 +1,560 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package v2alpha1
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// spec:
+//   targetRef:
+//     apiVersion: apps/v1
+//     kind: Deployment
+//     name: test
+//   owner: local
+//   remoteVersion: 1
+//   actuation:
+//     mode: Apply | Preview
+//     update:
+//       strategy: Auto|Disabled
+//     upscale:
+//       strategy: Max|Min|Disabled
+//       rules:
+//         - type: Pods|Percent
+//           value: 1
+//           periodSeconds: 60
+//     downscale:
+//       strategy: Max|Min|Disabled
+//       rules:
+//         - type: Pods|Percent
+//           value: 1
+//           periodSeconds: 60
+//   objectives:
+//     - type: PodResource
+//       resource:
+//         name: cpu
+//         value:
+//           type: Absolute|Utilization
+//           absolute: 500m
+//           utilization: 80
+//     - type: ContainerResource
+//       containerResource:
+//         name: cpu
+//         value:
+//           type: Absolute|Utilization
+//           absolute: 500m
+//           utilization: 80
+//   constraints:
+//     minReplicas: 1
+//     maxReplicas: 10
+//     containers:
+//       - name: "*"
+//         enabled: true
+//         requests:
+//           minAllowed:
+//           maxAllowed:
+
+// DatadogPodAutoscalerOwner defines the source of truth for this object (local or remote)
+// +kubebuilder:validation:Enum:=Local;Remote
+type DatadogPodAutoscalerOwner string
+
+const (
+	// DatadogPodAutoscalerLocalOwner states that this `DatadogPodAutoscaler` object is created/managed outside of Datadog app.
+	DatadogPodAutoscalerLocalOwner DatadogPodAutoscalerOwner = "Local"
+
+	// DatadogPodAutoscalerLocalOwner states that this `DatadogPodAutoscaler` object is created/managed in Datadog app.
+	DatadogPodAutoscalerRemoteOwner DatadogPodAutoscalerOwner = "Remote"
+)
+
+// DatadogPodAutoscalerSpec defines the desired state of DatadogPodAutoscaler
+type DatadogPodAutoscalerSpec struct {
+	// TargetRef is the reference to the resource to scale.
+	TargetRef autoscalingv2.CrossVersionObjectReference `json:"targetRef"`
+
+	// Owner defines the source of truth for this object (local or remote)
+	// Value needs to be set when a DatadogPodAutoscaler object is created.
+	Owner DatadogPodAutoscalerOwner `json:"owner"`
+
+	// RemoteVersion is the version of the .Spec currently store in this object.
+	// Only set if the owner is Remote.
+	RemoteVersion *uint64 `json:"remoteVersion,omitempty"`
+
+	// Actuation defines how recommendations should be applied.
+	// +optional
+	// +kubebuilder:default={}
+	Actuation *DatadogPodAutoscalerActuation `json:"actuation,omitempty"`
```

Comment:
Still not a big fan of presenting the `actuation` name to customers but LGTM if we don't find anything else.

Reply:
Some other ideas:
- `ActionPolicy` - what actions the autoscaler is allowed to take
- `ApplyPolicy` - what changes the autoscaler is allowed to apply
- `UpdatePolicy` - what updates the autoscaler is allowed to execute

I'm partial to `Actuation` since I'm very familiar with the term but most users won't be so I think its worth some thought to pick an expressive name here.

Reply:
I prefer the term actuation. I think it's clear and isn't overloaded.

Reply:
We’ve decided to go with `ApplyPolicy` as the field name. This choice provides clarity and aligns with how customers describe recommendation propagation.